src/fi_tostr.c

```diff
@@ -122,26 +122,26 @@ static void ofi_tostr_addr_format(char *buf, uint32_t addr_format)
 	CASEENUMSTR(FI_ADDR_EFA);
 	default:
 		if (addr_format & FI_PROV_SPECIFIC)
-			ofi_strcatf(buf, "Provider specific");
+			ofi_strcatf(buf, buflen, "Provider specific");
 		else
-			ofi_strcatf(buf, "Unknown");
+			ofi_strcatf(buf, buflen, "Unknown");
 		break;
 	}
 }
 
-static void ofi_tostr_progress(char *buf, enum fi_progress progress)
+static void ofi_tostr_progress(char *buf, size_t buflen, enum fi_progress progress)
```

Comment:
Does CASENUMSTR need to be updated as well?  The buflen is only used in the unknown case now.

Reply:
`CASENUMSTR` is declared the same way as `IFFLAGSTR` and implicitly use `buf` and `buflen`:

```
 #define CASEENUMSTR(SYM) \
       case SYM: { ofi_strncatf(buf, buflen, #SYM); break; }
```